### PR TITLE
tag.Add: allows users to register custom or missing tags into the tags registry

### PR DESCRIPTION
**Pull request by flicaflow**
From the discussion of the unmerged and probably obsolete PR #271 the idea was to Just a RegisterCustom function to the tag package. This PR is exactly that. I needed the TrackingID and TrackingUID tags which are currently not available. Following code enables me to use them now:

```
var (
	TagTrackingID  = tag.Tag{Group: 0x0062, Element: 0x0020}
	TagTrackingUID = tag.Tag{Group: 0x0062, Element: 0x0021}
)

func init() {
	tag.RegisterCustom(tag.Info{
		Tag:  TagTrackingID,
		VR:   "UT",
		Name: "TrackingID",
		VM:   "1",
	})
	tag.RegisterCustom(tag.Info{
		Tag:  TagTrackingUID,
		VR:   "UI",
		Name: "TrackingUID",
		VM:   "1",
	})
}
```

### CODE REVIEW

**Comment by suyashkumar** [comment]
Thank you @flicaflow! Couple thoughts:
* Can you add a couple sanity tests along with this? e.g. register a new tag and then check that you can successfully find it via one of the tag package APIs? 
* How should we handle cases where a tag a user is trying to add already exists in the underlying dict? Should we return an error unless the user also sets an `overrideExisting = true` or something?

Otherwise, having a feature like this sounds good to me! :)

**Review by suyashkumar — COMMENTED** [review]
Thank you @flicaflow! Couple thoughts:
* Can you add a couple sanity tests along with this? e.g. register a new tag and then check that you can successfully find it via one of the tag package APIs? (maybe in tag_test.go).
* How should we handle cases where a tag a user is trying to add already exists in the underlying dict? Should we return an error unless the user also sets an `overrideExisting = true` or something?

Otherwise, having a feature like this sounds good to me! :)

**Comment by flicaflow** [comment]
Hi @suyashkumar ,
* I agree, this code needs a unit test. I will add that.
* I as a user of the API would expect that Registering a tag will just do that and eventually overriding an existing one. Maybe that is even desired because you need to give it another VR type. Having this not as the default is dangerous because maybe with an update the list of pre-defined tags gets extended and suddenly my code breaks. Users can easily add a check to see if a tag already exists if desired. If we want to have this feature in it I would vote for a second function like `RegisterCustomIfNotExist(...)` 

Thanks for you reply, the unit test follows soon.

File: pkg/tag/tag.go [hunk 1]
```diff
@@ -231,3 +231,12 @@ func parseTag(tag string) (Tag, error) {
 	}
 	return Tag{Group: uint16(group), Element: uint16(elem)}, nil
 }
+
+// RegisterCustom allows to add a custom tag. This allows to work arond missing tag definitions
+// and to create private tags.
```
Comment (suyashkumar) [inline pkg/tag/tag.go:236]:
```suggestion
// RegisterCustom adds a custom tag to this package. This enables users to work around missing tag definitions
// and to create private tags.
```

File: pkg/tag/tag_test.go [hunk 2]
```diff
@@ -45,6 +45,70 @@ func TestSplitTag(t *testing.T) {
...
+		}
+		info, err := Find(TagTestTag)
+		if err != nil {
+			t.Fatalf("expected TestTag to be accessible with Find")
+		}
+		if info.VR != "UT" ||
+			info.Name != "TestTag" ||
+			info.VM != "1" {
+			t.Fatal("info of new registered tag is wrong")
+		}
```
Comment (suyashkumar) [inline pkg/tag/tag_test.go:79]: nit: you should be able to use `cmp.Diff` to check the info exactly matches the struct you created above: https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff

You could optionally check that both FindByName and Find returns the correct struct.

File: pkg/tag/tag_test.go [hunk 3]
```diff
@@ -45,6 +45,70 @@ func TestSplitTag(t *testing.T) {
...
+			t.Fatalf("expected TestTag to be accessible with FindByName")
+		}
+		info, err := Find(TagTestTag)
+		if err != nil {
+			t.Fatalf("expected TestTag to be accessible with Find")
+		}
+		if info.VR != "UT" ||
+			info.Name != "TestTag" ||
+			info.VM != "1" {
+			t.Fatal("info of new registered tag is wrong")
```
Comment (suyashkumar) [inline pkg/tag/tag_test.go:78]: nit: prefer `t.Errorf` here and above to collect all test errors. Typically I use t.Errorf for test failure conditions, and Fatalf for test setup issues where the test needs to stop immediately.

**Comment by flicaflow** [comment]
I absolutely agree with naming it `SetInfo`. I used `RegisterCustom` because you suggested it in the other PR long time ago. I also think the force flag is a good way to communicate what is happening. 
In that case We could also just call it `Set` or even `Add`. This makes sense int the context of the tag package (`tag.Set(...)`) . Maybe `Add` does convey more the normal behavior (adding new tags) and the `force` flag makes it clear what is happening on collisions.
```go
err := tag.Add(newTag, false) // errors if the tag already exists

err = tag.Add(newTag, true) // overwrites an eventually existing tag and never returns an error
```

**Comment by suyashkumar** [comment]
Makes sense, and I think that sounds good to me. `tag.Add(info tag.Info, force bool)` seems reasonable to me! It's a bit extra API than I'd usually prefer by adding the force flag, but I think in our case it can be good to force the user to decide and think about if they'd always want to override a colliding tag or not. Probably not a big deal either way, so happy to move forward with that! Thank you! :)

**Review by suyashkumar — APPROVED** [review]
LGTM, with one minor typo. I'm not sure that I can commit to this branch otherwise I'd have sent the fix myself!

**Comment by suyashkumar** [comment]
Oh, jk, looks like it allowed me to add a commit!

**Comment by suyashkumar** [comment]
Merged 🎉 ! Thank you for the contribution @flicaflow :).

**Pull request merged**